Crypto/PubKey/Ed448.hs

```diff
@@ -105,8 +106,27 @@ toPublic (SecretKey sec) = PublicKey
 {-# NOINLINE toPublic #-}
 
 -- | Sign a message using the key pair
+--   The public key parameter is ignored and its public key
+--   is generated from the secret key parameter to prevent
+--   Double Public Key Signing Function Oracle Attack.
 sign :: ByteArrayAccess ba => SecretKey -> PublicKey -> ba -> Signature
-sign secret public message =
+sign secret _public message =
+    Signature $ B.allocAndFreeze signatureSize $ \sig ->
+        withByteArray secret $ \sec ->
+            withByteArray public $ \pub ->
+                withByteArray message $ \msg ->
+                    decaf_ed448_sign sig sec pub msg (fromIntegral msgLen) 0 no_context 0
```

Comment:
@kazu-yamamoto FYI the cbits for ed448 already rederive the public key and perform an equality check (`abort()` on mismatch).

As a result of this change, `sign` will now rederive the public key two times.  With these changes in place on the Haskell side, I would consider removing the rederivation from the C code (you could promote `crypton_decaf_ed448_sign_internal` to replace `crypton_decaf_ed448_sign`).  This would avoid the double-rederive in `sign` and restore the performance advantage (and potential of unsafe use) of `unsafeSign`.

Reply:
Thank you for informing this.
May I ask you to send a PR?

Reply:
Sure, I'll create it later today.

Reply:
There would be no urgency to cut a new release, unless someone complains about the performance :)

Reply:
Here you go: https://github.com/kazu-yamamoto/crypton/pull/48